vllm/model_executor/layers/quantization/tpu_int8.py

```diff
@@ -0,0 +1,119 @@
+from typing import Any, Dict, List, Optional, Tuple
+
+import torch
+from torch.nn import Module
+from torch.nn.parameter import Parameter
+
+from vllm.model_executor.layers.linear import LinearBase, LinearMethodBase
+from vllm.model_executor.layers.quantization.base_config import (
+    QuantizationConfig)
+from vllm.model_executor.utils import set_weight_attrs
+
+ACTIVATION_SCHEMES = ["none"]
+
+
+class Int8TpuConfig(QuantizationConfig):
+    """Int8 Quantization Config class for TPU Backend.
+    """
+
+    def __init__(
+        self,
+        activation_scheme: str = "none",
+    ) -> None:
+        if activation_scheme not in ACTIVATION_SCHEMES:
+            raise ValueError(
+                f"Unsupported activation scheme {activation_scheme}")
+        self.activation_scheme = activation_scheme
+
+    def get_name(self) -> str:
+        return "tpu_int8"
+
+    def get_supported_act_dtypes(self) -> List[torch.dtype]:
+        return [torch.float16, torch.bfloat16]
+
+    @classmethod
+    def get_min_capability(cls) -> int:
+        raise NotImplementedError(
+            "This function should not be called with TPU Backend")
+
+    @staticmethod
+    def get_config_filenames() -> List[str]:
+        return []
+
+    @classmethod
+    def from_config(cls, config: Dict[str, Any]) -> "Int8TpuConfig":
+        activation_scheme = cls.get_from_keys(config, ["activation_scheme"])
+        return cls(activation_scheme=activation_scheme)
+
+    def get_quant_method(self, layer: Module,
+                         prefix: str) -> Optional["TPUInt8LinearMethod"]:
+        if isinstance(layer, LinearBase):
+            return TPUInt8LinearMethod(self)
+        return None
+
+    def get_scaled_act_names(self) -> List[str]:
+        return []
+
+
+class TPUInt8LinearMethod(LinearMethodBase):
+    """Int8 Linear method for TPU Quant.
+
+    """
+
+    def __init__(self, quant_config: Int8TpuConfig):
+        self.quant_config = quant_config
+
+    def create_weights(self, layer: Module, input_size_per_partition: int,
+                       output_partition_sizes: List[int], input_size: int,
+                       output_size: int, params_dtype: torch.dtype,
+                       **extra_weight_attrs):
+        try:
+            import torch_xla.experimental.xla_quantized_matmul  # noqa: F401
+        except ImportError as err:
+            raise ImportError(
+                "Please install torch_xla by following the instructions at "
+                "https://docs.vllm.ai/en/latest/getting_started/tpu-installation.html "  # noqa: E501
+                "to run vLLM on TPU.") from err
```

Comment:
QQ: Why is this import statement needed here?

Reply:
This import will register the xla quantized op under `torch.ops` [ref](https://chat.google.com/room/AAAA33ZmXmY/3J3BHfmprCg/T_7qL5Rd6t0?cls=10), in this way it will be traceable by `torch.dynamo`.

Then, we can call the quantized op by `torch.ops.xla.quantized_matmul` [here](https://github.com/vllm-project/vllm/pull/7005/files#diff-54e999a436dc0eb66e02ae3680278ba8f48965e5ae324ff18c24f3a0646a8193R119)

Reply:
But why can this not be imported at the top of the file?

Reply:
Hi @robertgshaw2-neuralmagic, it's because all configs are imported in https://github.com/vllm-project/vllm/blob/main/vllm/model_executor/layers/quantization/__init__.py. If we put it at the top of the file, we will hit import failure for non-tpu case. The current way to import is similar to deepspeed [one](https://github.com/vllm-project/vllm/blob/460c1884e3cb781730f85cb5591a85d5864bdac8/vllm/model_executor/layers/quantization/deepspeedfp.py#L144-L153)

Reply:
Why don't we import it in the `apply` method?

Reply:
Yeah, makes sense, updated.